Download list from source page

emoryr opened the pull request:
Download the newest list from https://semsa.manaus.am.gov.br/sala-de-situacao/novo-coronavirus/

Issue #13

Signed-off-by: Emory Freitas <emory.rvf@gmail.com>

emoryr committed:
Download list from source page

Download the newest list from https://semsa.manaus.am.gov.br/sala-de-situacao/novo-coronavirus/

Issue #13

Signed-off-by: Emory Freitas <emory.rvf@gmail.com>

pvfrota commented on requirements.txt:
```diff
@@ -10,4 +10,6 @@ pytz==2020.5
-Wand==0.6.5
\ No newline at end of file
+Wand==0.6.5
+requests
```
@emoryr Obrigado pela contribuição :1st_place_medal: 

Acho q seria bom fazer o lock das versões, seguindo o padrão

Além disso, manter os nomes das libs em ordem alfabética

emoryr replied:
PR atualizado

emoryr committed:
Lock lib version

Signed-off-by: Emory Freitas <emory.rvf@gmail.com>

pvfrota commented:
Parabéns, eu testei e funcionou tudo direitinho :100: 

Apenas adicionar algumas melhorias que eu observei durante os testes (inclusive foi mal não ter testado e dado feedback de uma vez):

- Como a gente tem que manter o histórico de arquivos, o ideal seria manter o nome original do arquivo, pra não sobrescrever a cada dia.

- Mudar o nome do arquivo Python para `download_data.py`, para seguir o padrão.


- Adicionar a task de download dos dados ao `Makefile`, atualizar a task `data` para incluí-la:

```Makefile

...

.PHONY: data
data: download-data extract-data process-data

.PHONY: download-data
download-data:
	@docker-compose run --rm ${SERVICE_NAME} python download_data.py

...

```

- Organizar o arquivo `requirements.txt` por ordem alfabética (my bad, eu não vi que já estava desorganizado anteriormente ao seu PR)

gilvan-reis commented on extract_page.py:
```diff
@@ -0,0 +1,29 @@
+from bs4 import BeautifulSoup
+
+
+# Set the URL you want to webscrape from
```
Não impeditivo para o merge: Na minha opinião, esse comentários podem ser removidos visto que eles apenas explicam o que é feito no código. Comentários são melhores aproveitados quando explicam porque algo é feito (o como já está descrito no código)

emoryr committed:
Change request by reviews

 - Rename extract_page.py to download_page.py
 - Remove unnecessary comments
 - Remove unnecessary code
 - Keep the file name downloaded

Signed-off-by: Emory Freitas <emory.rvf@gmail.com>

pvfrota commented on Makefile:
```diff
@@ -16,6 +16,10 @@ build-no-cache:
 .PHONY: data
 data: extract-data process-data
```
:warning:  Não esquecer de adicionar a task `download-data` à task `data`

> * Adicionar a task de download dos dados ao `Makefile`, atualizar a task `data` para incluí-la:
> 
> ```makefile
> ...
> 
> .PHONY: data
> data: download-data extract-data process-data
> 
> .PHONY: download-data
> download-data:
> 	@docker-compose run --rm ${SERVICE_NAME} python download_data.py
> 
> ...
> ```

```suggestion
data: download-data extract-data process-data
```

Só faltou esse pequeno detalhe, mas de resto, aprovado. Valeu! :dancing_men:

emoryr committed:
Adding download-data to data target

Signed-off-by: Emory Freitas <emory.rvf@gmail.com>

pvfrota commented:
![](https://media1.tenor.com/images/d4b72b2c60faea27d5f5b4d357b28a05/tenor.gif?itemid=11326155)
